File: edge/src/greenhouse/ToggleLed.cpp
#include "Led.h"
#include "ToggleLed.h"

ToggleLed::ToggleLed(int pin) {
  this->isOn = false;
  this->light = new Led(pin);
}

void ToggleLed::switchOn() {
  light->switchOn();
}

void ToggleLed::switchOff() {
  light->switchOff();
}

void ToggleLed::toggle() {
  if(isOn) {
    isOn = false;
    light->switchOff();
  } else {
    isOn = true;
    light->switchOn();
  }
}

File: hub/src/display/ToggleLed.java
package display;

import java.io.IOException;

public class ToggleLed implements ToggleLight {

	private Light led;
	private Boolean isOn;
	
	public ToggleLed(int pinNum) {
		this.led = new Led(pinNum);
		this.isOn = false;
	}

	@Override
	public synchronized void toggle() throws IOException {
		if(isOn) {
			led.switchOff();
		} else {
			led.switchOn();
		}
	}

	@Override
	public synchronized void switchOn() throws IOException {
		led.switchOn();
	}

	@Override
	public synchronized void switchOff() throws IOException {
		led.switchOff();
	}

}
